client/lib/products-values/constants.js

```diff
@@ -181,10 +189,55 @@ export const getJetpackProducts = () => {
 				yearly: [ PRODUCT_JETPACK_SEARCH ],
 				monthly: [ PRODUCT_JETPACK_SEARCH_MONTHLY ],
 			},
-			optionShortNames: getJetpackProductsShortNames(),
+			optionShortNamesCallback: productObject => {
+				const numberOfDefinedTiers = 5;
+				switch ( productObject.price_tier_slug ) {
+					case JETPACK_SEARCH_TIER_UP_TO_100_RECORDS:
+						return translate( 'Tier 1: Up to 100 records' );
+					case JETPACK_SEARCH_TIER_UP_TO_1K_RECORDS:
+						return translate( 'Tier 2: Up to 1,000 records' );
+					case JETPACK_SEARCH_TIER_UP_TO_10K_RECORDS:
+						return translate( 'Tier 3: Up to 10,000 records' );
+					case JETPACK_SEARCH_TIER_UP_TO_100K_RECORDS:
+						return translate( 'Tier 4: Up to 100,000 records' );
+					case JETPACK_SEARCH_TIER_UP_TO_1M_RECORDS:
+						return translate( 'Tier 5: Up to 1,000,000 records' );
+					case JETPACK_SEARCH_TIER_MORE_THAN_1M_RECORDS: {
+						// This is a catch-all tier with prices increasing
+						// proportionally per million records, so define fake
+						// tiers here to show the user what they will actually
+						// pay and why.
```

Comment:
Not entirely sure if this is how we want to display things for sites with 1,000,000+ records, but it seemed like the simplest way to do it for now.

Reply:
In general I wouldn't expose the tier numbers to end users, but instead just say what they can go up to. But that is all design and copy I think so this looks fine as is.

Reply:
Yeah, I was just copying it from the older designs -- it can definitely be changed.  I guess maybe the idea was that it's useful to get the concept of different "tiers" into the text somehow, though, so the user can see how we're charging for things.